commit 32ab0a38f0bd554cc45203ff4fdb6b0fdea6f025
Merge: 976990569360 514877182b53
Author: David S. Miller <davem@davemloft.net>
Date:   Sat Oct 29 17:28:45 2016 -0400

    Merge tag 'mac80211-next-for-davem-2016-10-28' of git://git.kernel.org/pub/scm/linux/kernel/git/jberg/mac80211-next
    
    Johannes Berg says:
    
    ====================
    Among various cleanups and improvements, we have the following:
     * client FILS authentication support in mac80211 (Jouni)
     * AP/VLAN multicast improvements (Michael Braun)
     * config/advertising support for differing beacon intervals on
       multiple virtual interfaces (Purushottam Kushwaha, myself)
     * deprecate the old WDS mode for cfg80211-based drivers, the
       mode is hardly usable since it doesn't support any "modern"
       features like WPA encryption (2003), HT (2009) or VHT (2014),
       I'm not even sure WEP (introduced in 1997) could be done.
    ====================
    
    Signed-off-by: David S. Miller <davem@davemloft.net>

diff --cc net/wireless/nl80211.c
index 0d3ab4bfeacf,bb30fa1969f9..24ab199ef2fc
--- a/net/wireless/nl80211.c
+++ b/net/wireless/nl80211.c
@@@ -12583,23 -12692,16 +12676,31 @@@ static const struct genl_ops nl80211_op
  		.internal_flags = NL80211_FLAG_NEED_NETDEV_UP |
  				  NL80211_FLAG_NEED_RTNL,
  	},
+ 	{
+ 		.cmd = NL80211_CMD_SET_MULTICAST_TO_UNICAST,
+ 		.doit = nl80211_set_multicast_to_unicast,
+ 		.policy = nl80211_policy,
+ 		.flags = GENL_UNS_ADMIN_PERM,
+ 		.internal_flags = NL80211_FLAG_NEED_NETDEV |
+ 				  NL80211_FLAG_NEED_RTNL,
+ 	},
  };
  
 +static struct genl_family nl80211_fam __ro_after_init = {
 +	.name = NL80211_GENL_NAME,	/* have users key off the name instead */
 +	.hdrsize = 0,			/* no private header */
 +	.version = 1,			/* no particular meaning now */
 +	.maxattr = NL80211_ATTR_MAX,
 +	.netnsok = true,
 +	.pre_doit = nl80211_pre_doit,
 +	.post_doit = nl80211_post_doit,
 +	.module = THIS_MODULE,
 +	.ops = nl80211_ops,
 +	.n_ops = ARRAY_SIZE(nl80211_ops),
 +	.mcgrps = nl80211_mcgrps,
 +	.n_mcgrps = ARRAY_SIZE(nl80211_mcgrps),
 +};
 +
  /* notification functions */
  
  void nl80211_notify_wiphy(struct cfg80211_registered_device *rdev,